File: app/src/main/java/ng/com/techdepo/esidemtest/database/AppDatabase.java
package ng.com.techdepo.esidemtest.database;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;
import android.content.Context;

import ng.com.techdepo.esidemtest.utils.Converter;


@Database(entities = {QuestionEntity.class,Result.class},version = 1,exportSchema = false)
@TypeConverters({Converter.class})
public abstract class AppDatabase extends RoomDatabase {

    private static final String LOG_TAG = AppDatabase.class.getSimpleName();
    private static final Object LOCK = new Object();
    private static final String DATABASE_NAME = "questionlist";
    private static AppDatabase sInstance;

    public static AppDatabase getInstance(Context context) {
        if (sInstance == null) {
            synchronized (LOCK) {

                sInstance = Room.databaseBuilder(context.getApplicationContext(),
                        AppDatabase.class, AppDatabase.DATABASE_NAME)
                        .build();
            }
        }

        return sInstance;
    }

    public abstract DatabaseDAO databaseDAO();
    public abstract ResultDAO resultDAO();

}

File: app/src/main/java/ng/com/techdepo/esidemtest/database/AppDatabase.kt
package ng.com.techdepo.esidemtest.database

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import ng.com.techdepo.esidemtest.utils.Converter


@Database(entities = arrayOf(QuestionEntity::class, Result::class), version = 1, exportSchema = false)
@TypeConverters(Converter::class)
abstract class AppDatabase : RoomDatabase() {

    abstract fun databaseDAO(): DatabaseDAO
    abstract fun resultDAO(): ResultDAO

}
